io.openems.edge.greenconsumptionadvisor.api/src/io/openems/edge/greenconsumptionadvisor/api/GreenConsumptionAdvisorImpl.java

```diff
@@ -0,0 +1,193 @@
+package io.openems.edge.greenconsumptionadvisor.api;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.temporal.ChronoUnit;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.metatype.annotations.Designate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import io.openems.common.utils.JsonUtils;
+import io.openems.common.exceptions.OpenemsError.OpenemsNamedException;
+import io.openems.edge.common.component.AbstractOpenemsComponent;
+import io.openems.edge.common.component.OpenemsComponent;
+import io.openems.edge.controller.api.Controller;
+
+@Designate(ocd = Config.class, factory = true)
+@Component(//
+		name = "Green.Consumption.Advice.API", // This name has to be kept for compatibility reasons
+		immediate = true, //
+		configurationPolicy = ConfigurationPolicy.REQUIRE //
+)
+public class GreenConsumptionAdvisorImpl extends AbstractOpenemsComponent
+		implements Controller, OpenemsComponent, GreenConsumptionAdvisor {
+
+	@Reference
+	private ConfigurationAdmin cm;
+
+	private final Logger log = LoggerFactory.getLogger(GreenConsumptionAdvisorImpl.class);
+
+	private static final String URL_API_RECOMMONDATION = "https://api.corrently.io/v2.0/gsi/advisor?zip=";
```

Comment:
The method name writeRecommondationToChannel contains a typo ("Recommondation" should be "Recommendation")